### DISCUSSION TITLE
Error: CLI arg for `appdir` is not declared for this command

### DISCUSSION THREAD

**Discussion by mwarkentin** [Homebrew/discussions#1105] [Everyday usage] [answered]
I'm getting the following error when running `brew list`:

```
❯ brew list
adns			cfssl			git-chglog		heroku			libksba			mysql			packer			python@3.9		tfenv
...
ccat			git			hadolint		libidn2			mtr			pack			python@3.8		terraform_landscape	youtube-dl
Error: CLI arg for `appdir` is not declared for this command
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/cli/args.rb:147:in `method_missing'
/usr/local/Homebrew/Library/Homebrew/cask/config.rb:45:in `from_args'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/list.rb:45:in `list_casks'
/usr/local/Homebrew/Library/Homebrew/cmd/list.rb:158:in `list_casks'
/usr/local/Homebrew/Library/Homebrew/cmd/list.rb:120:in `list'
/usr/local/Homebrew/Library/Homebrew/brew.rb:122:in `<main>'
```

Anyone seen this before? The command seems to work fine, but we have some scripting which uses this and the non-0 exit code is breaking things.

**Comment by MxHex** [comment]
This is linked to my issue -> https://github.com/Homebrew/discussions/discussions/1104
I'm experiencing this as well since today after the update.

**Reply by carlocab** [reply]
What's the output of `brew config` and `brew doctor`?

**Reply by MxHex** [reply]
> ▶ brew config
HOMEBREW_VERSION: 3.0.8-6-ga235b38
ORIGIN: https://github.com/Homebrew/brew
HEAD: a235b38000d06b6732dcf1d5deab7ad57710ad11
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core.git
Core tap HEAD: 3f7c96754c6767921ca209bea16c88e944c1feab
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: N/A
Git: 2.30.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: N/A
Xcode: N/A

> ▶ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

> Warning: Unknown support status

> Warning: Some installed kegs have no formulae!
This means they were either deleted or installed with `brew diy`.
You should find replacements for the following formulae:
  node@8
  python@2

> Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  dshb

> Warning: No developer tools installed.
Install the Command Line Tools:
  xcode-select --install


> Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  docker
  python@2
  openssh

**Comment by Bo98** [comment]
https://github.com/Homebrew/brew/pull/10905 will fix this.

**Reply by MxHex** [reply]
I seems to be a wider issue though. Did you checkout my issue? https://github.com/Homebrew/discussions/discussions/1104

**Reply by Bo98** [reply]
I'm handling your issue in a separate pull request: https://github.com/Homebrew/brew/pull/10904. Your issue only affected those without dev tools installed.

**Comment by carlocab** [comment, marked as answer]
Should be fixed in https://github.com/Homebrew/brew/pull/10905.

**Reply by mwarkentin** [reply]
Confirmed fixed, thanks!
